src/MICore/CommandFactories/lldb.cs

```diff
@@ -46,12 +46,15 @@ public override bool SupportsFrameFormatting
             }
         }
 
-        protected override StringBuilder BuildBreakInsert(string condition, bool enabled)
+        protected async override Task<StringBuilder> BuildBreakInsert(string condition, bool enabled)
         {
-            // LLDB's use of the pending flag requires an optional parameter or else it fails.
+            // LLDB's 3.5 use of the pending flag requires an optional parameter or else it fails.
             // We will use "on" for now. 
-            // TODO: Fix this on LLDB-MI's side
-            string pendingFlag = "-f on ";
+            string pendingFlag = "-f ";
+            if (await RequiresOnKeywordForBreakInsert())
+            {
+                pendingFlag = "-f on ";
```

Comment:
Create a `const string pendingFlagParameter = "on";`  and after the `cmd.Append(pendingFlag);` check `RequiresOnKeywordForBreakInsert()` and if it comes back as true, then append `pendingFlagParameter`.